### PR TITLE
Drop support for EOL Python 3.4 and 3.5

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/morepath/morepath/issues/544.

* First, fix Flake8 and format with Black to get the CI back to green
* Drop support for EOL Python 3.4 and 3.5
* Upgrade Python syntax with `pyupgrade --py36-plus`
* Remove unused imports
* Also a nit: drop the dot in py.test: https://twitter.com/pytestdotorg/status/753767547866972160

### CODE REVIEW

File: morepath/tests/test_path_directive.py [hunk 1]
```diff
@@ -1802,7 +1800,7 @@ def link(self, request):
     c = Client(App())
 
     response = c.get("/?s=sim~ple")
-    assert response.body == "View: sim~ple".encode("utf-8")
+    assert response.body == b"View: sim~ple"
```
Comment (jugmac00) [inline morepath/tests/test_path_directive.py:1803]: This was new to me! Looks good: https://github.com/asottile/pyupgrade/issues/138

**Comment by jugmac00** [comment]
Thank you very much!

I am curious - do you use `Morepath` yourself?

P.S.: When I run `tox -e pep8`, flake8 tries to lint all the development checkouts in the `src` directory.

```
pep8 run-test: commands[0] | flake8
./src/dectate/dectate/toposort.py:4:22: E741 ambiguous variable name 'l'
def topological_sort(l, get_depends):
                     ^
./src/reg/reg/predicate.py:60:9: E731 do not assign a lambda expression, use a def
        get_key = lambda d: func(**d)
        ^
./src/reg/reg/predicate.py:79:9: E731 do not assign a lambda expression, use a def
        get_key = lambda d: d[name].__class__
        ^
./src/reg/reg/predicate.py:81:9: E731 do not assign a lambda expression, use a def
        get_key = lambda d: func(**d).__class__
        ^
./src/reg/reg/predicate.py:102:9: E731 do not assign a lambda expression, use a def
        get_key = lambda d: func(**d)
        ^
./src/reg/reg/tests/test_docgen.py:8:37: E741 ambiguous variable name 'l'
    return "\n".join(l.rstrip() for l in s.splitlines())
                                    ^
ERROR: InvocationError for command /home/jugmac00/Projects/morepath/.tox/pep8/bin/flake8 (exited with code 1)
_____________________________________________________ summary _____________________________________________________
ERROR:   pep8: commands failed
```
see https://morepath.readthedocs.io/en/latest/developing.html#install-morepath-for-development

I will create another PR to explicitly pass `morepath/ setup.py` as path argument.

**Comment by hugovk** [comment]
No, I've not used it, just saw an issue I thought I could help out with :)

Thanks for the Flake8 fix!

**Comment by jugmac00** [comment]
Awesome! :-D

**Pull request merged**